app/code/Magento/AwsS3/Driver/AwsS3Factory.php

```diff
@@ -158,7 +158,7 @@ public function createConfigured(
                 'cache' => $cache
             ]
         );
-        $objectUrl = rtrim($client->getObjectUrl($config['bucket'], './'), '/') . trim($prefix, '\\/') . '/';
+        $objectUrl = rtrim($client->getObjectUrl($config['bucket'], './'), '/') . '/' . trim($prefix, '\\/') . '/';
```

Comment:
When prefix is empty this produces a double slash (//) in the URL. Consider only adding the slash when prefix is non-empty, for example: ($prefix !== '' ? '/' . trim($prefix, '\/') : '').